sdk/core/core/inc/az_auth.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_AUTH_H
+#define AZ_AUTH_H
+
+#include <az_mut_span.h>
+#include <az_pair.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include <stdbool.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef enum {
+  AZ_AUTH_KIND_CLIENT_CREDENTIALS = 1,
+} az_auth_kind;
+
+typedef struct {
+  az_span tenant_id;
+  az_auth_kind kind;
```

Comment:
As we discussed yesterday, this implementation is using discriminated unions. I would confirm with @JeffreyRichter which options we would like to go long term and what's Ok for short term.

Reply:
Sure, this is the starting point with a simple straightfoward implementation, let's start from here.
Per our standup conversation yesterday we all rather agreed, that using callback was not neccessary at this point, given that I need buffer, and http cliend inside the callback. But later we'll need to add token cache, interactive logon, and the design might need to become more complex.